e2e-tests/tfm/cypress/e2e/journeys/case-amendments/facility-end-date/feature-flag-enabled/gef-deal-add-multiple-amendments-impacting-facility-end-date.spec.js

```diff
@@ -0,0 +1,228 @@
+import { format } from 'date-fns';
+import relative from '../../../../relativeURL';
+import facilityPage from '../../../../pages/facilityPage';
+import { ADMIN, BANK1_MAKER1, PIM_USER_1, T1_USER_1 } from '../../../../../../../e2e-fixtures';
+import { MOCK_FACILITY_ONE } from '../../../../../fixtures/mock-gef-facilities';
+import { MOCK_APPLICATION_AIN } from '../../../../../fixtures/mock-gef-deals';
+import { DEAL_TYPE } from '../../../../../../../gef/cypress/fixtures/constants';
+import amendmentsPage from '../../../../pages/amendments/amendmentsPage';
+import { todayDay, todayMonth, todayYear } from '../../../../../../../e2e-fixtures/dateConstants';
+
+if (Cypress.env('FF_TFM_FACILITY_END_DATE_ENABLED') === 'true') {
+  context('Amendments - GEF deal add multiple consecutive amendments impacting facility end date values', () => {
+    let dealId;
+    let facility;
+
+    const Date1 = new Date('2024-01-01');
+    const Date2 = new Date('2024-02-02');
+    const Date3 = new Date('2024-03-03');
+    const Date4 = new Date('2024-04-04');
+    const Date5 = new Date('2024-05-05');
+
+    const currentDateFormat = 'dd MMMM yyyy';
+    const checkAnswersDateFormat = 'dd MMM yyyy';
+    const facilityPageDateFormat = 'd MMMM yyyy';
+
+    const MOCK_GEF_FACILITY_WITH_BANK_REVIEW_DATE = {
+      ...MOCK_FACILITY_ONE,
+      hasBeenIssued: true,
+      isUsingFacilityEndDate: false,
+      facilityEndDate: undefined,
+      bankReviewDate: Date1,
+    };
+
+    before(() => {
+      // inserts a gef deal
+      cy.insertOneGefDeal(MOCK_APPLICATION_AIN, BANK1_MAKER1).then((insertedDeal) => {
+        dealId = insertedDeal._id;
+        // updates a gef deal so has relevant fields
+        cy.updateGefDeal(dealId, MOCK_APPLICATION_AIN, BANK1_MAKER1);
+
+        cy.createGefFacilities(dealId, [MOCK_GEF_FACILITY_WITH_BANK_REVIEW_DATE], BANK1_MAKER1).then((createdFacility) => {
+          facility = createdFacility.details;
+        });
+
+        cy.submitDeal(dealId, DEAL_TYPE.GEF, T1_USER_1);
+      });
+    });
+
+    beforeEach(() => {
+      cy.login(PIM_USER_1);
+      cy.visit(relative(`/case/${dealId}/facility/${facility._id}`));
+    });
+
+    after(() => {
+      cy.deleteDeals(dealId, ADMIN);
+    });
+
+    it('should display the current bank review date details from the facility snapshot in the Details tab', () => {
+      facilityPage.facilityIsUsingFacilityEndDate().should('have.text', 'No');
+      facilityPage.facilityBankReviewDate().should('have.text', format(Date1, facilityPageDateFormat));
+      facilityPage.facilityFacilityEndDate().should('not.exist');
+    });
+
+    it('should display the correct values during and after amending the bank review date for the first time', () => {
+      amendmentsPage.navigateToIsUsingFacilityEndDatePage({ startNewAmendment: true, newCoverEndDate: Date1 });
```

Comment:
these could go in a beforeEach on line 63 where we pass the date (line 65, 66 and 67) as repeated